lib/multicell_location/services/nrf_cloud_integration.c

```diff
@@ -60,6 +69,48 @@ const char *location_service_get_certificate_nrf_cloud(void)
 	return tls_certificate;
 }
 
+#if defined(CONFIG_NRF_CLOUD_MQTT)
+static void location_service_location_ready_cb(const struct nrf_cloud_cell_pos_result *const result)
+{
+	if (result != NULL) {
+		nrf_cloud_location.latitude = result->lat;
+		nrf_cloud_location.longitude = result->lon;
+		nrf_cloud_location.accuracy = (double)result->unc;
+	}
+	k_sem_give(&location_ready);
+}
+
+int location_service_get_cell_location_nrf_cloud(
+	const struct lte_lc_cells_info *cell_data,
+	char * const rcv_buf,
+	const size_t rcv_buf_len,
+	struct multicell_location *const location)
+{
+	ARG_UNUSED(rcv_buf);
+	ARG_UNUSED(rcv_buf_len);
+
+	int err;
+
+	err = nrf_cloud_cell_pos_request(cell_data, true, location_service_location_ready_cb);
+	if (err) {
+		LOG_ERR("Failed to request cell pos data, error: %d", err);
+		return err;
+	}
+
+	LOG_INF("Cellular positioning request sent");
+
+	if (k_sem_take(&location_ready, K_SECONDS(20)) == -EAGAIN) {
+		LOG_ERR("Cellular positioning data request timed out");
+		return -ETIMEDOUT;
+	}
+
+	location->latitude = nrf_cloud_location.latitude;
+	location->longitude = nrf_cloud_location.longitude;
+	location->accuracy = nrf_cloud_location.accuracy;
+
+	return err;
+}
+#else // defined(CONFIG_NRF_CLOUD_MQTT)
```

Comment:
As briefly discussed in Teams,  I'd like to have this implemented in other way.

I.e. enable the usage of both REST and MQTT with the same binary. That would be more user friendly and would enable easier service evaluation. 

The logic could be when both MQTT and REST are enabled in a build:
- at 1st check the MQTT connection status. not sure if there is a polling api for that one, but at least events are cloud events available.
- if cloud/MQTT is connected, continue with MQTT, otherwise shoot with the REST as previously.

Additionally, maybe there should be own Kconfig in multicell lib than using global: CONFIG_NRF_CLOUD_MQTT.

Reply:
This is aligned with A-GPS currently. Given the timeframe, can we really get more sophisticated stuff done and merged tomorrow. If yes, go ahead.

Reply:
I guess with PR https://github.com/nrfconnect/sdk-nrf/pull/6176, we could even first try MQTT and if it fails, use REST. And the same could be done for A-GPS. Another question is whether it's doable really quickly as  PR https://github.com/nrfconnect/sdk-nrf/pull/6176 won't get merged before tomorrow.

Reply:
We also need to consider whether this fallback is something that we should do. This sounds more like it's intended for our own testing than to be used by the customers. If something like this is implemented, it must be possible to disable the functionality. And it should probably also be disabled by default.

Reply:
This requires a bit more discussion with a broader audience for which there is no time now. We can consider this functionality for NCS 1.9.0. Resolving this discussion.